plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs

```diff
@@ -303,9 +306,25 @@ defineBuiltinTerms = do
     defineBuiltinTerm 'Builtins.appendString $ mkBuiltin PLC.AppendString
     defineBuiltinTerm 'Builtins.emptyString $ PIR.mkConstant () ("" :: Text)
     defineBuiltinTerm 'Builtins.equalsString $ mkBuiltin PLC.EqualsString
-    defineBuiltinTerm 'Builtins.trace $ mkBuiltin PLC.Trace
     defineBuiltinTerm 'Builtins.encodeUtf8 $ mkBuiltin PLC.EncodeUtf8
 
+    -- Tracing
+    defineBuiltinTerm 'Builtins.trace $
+        -- When `remove-trace` is specified, we define `trace` as `\_ a -> a` instead of the builtin version.
+        -- If simpilers are used, `trace t a` will be inlined and reduced to just `a`.
+        if coRemoveTrace compileOpts
+        then runQuote $ do
```

Comment:
```suggestion
        then liftQuote $ do
```
We've got `MonadQuote` here, I think.

Reply:
This call is in the second term argument of `defineBuiltinTerm` so `runQuote` is the one.

Reply:
You can just hoist the arg up:
```
traceTerm <- if ... then ... else
defineBuiltinTerm 'Builtins.traceTerm
```

Reply:
True, but I think the current code is scoped a bit better?

Reply:
Sure, but the extra `runQuote` might make the overall result not have globally unique names where otherwise it would, so it's somewhat costly. If you really care about scoping you can always put those in a nested do block.

Reply:
You're right, I totally missed that risk. Have pushed a new commit to address that.